tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+        akey = resp.json()["keys"][0]["access_key"]
+        skey = resp.json()["keys"][0]["secret_key"]
+        return akey, skey, user_id
+
+    def set_get_user_quota(self, config_dict, user_id):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        self.log.info("Perform get set user quota")
+        payload = self.csm_obj.iam_user_quota_payload(config_dict["enabled"],
+                                                      config_dict["max_size"],
+                                                      config_dict["max_objects"],
+                                                      check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(user_id, payload,
+                                                              verify_response=True)
+        assert result, f"Verification for get set user failed with error {resp}"
+
+    def verify_user_quota(self, akey, skey, user_id):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        total_objects = 0
+        total_size = 0
+        self.log.info("Get capacity count from AWS")
+        bkt_lst = s3_misc.list_bucket(akey, skey)
+        for bucket in bkt_lst:
+            objects, size = s3_misc.get_objects_size_bucket(bucket, akey, skey)
+            total_objects = total_objects + objects
+            total_size = total_size + size
+
+        self.log.info("total objects and size %s and %s ", total_objects, total_size)
+        self.log.info("Perform & Verify GET API to get capacity usage stats")
+        res, resp = self.csm_obj.verify_user_capacity(user_id, total_size,
+                                                      total_size, total_objects)
+        assert res, f"Verify User capacity failed with error msg {resp}"
+
+    def multipart_upload(self, bucket, obj, akey, skey, filesize):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        s3_mp_test_obj = S3MultipartTestLib(access_key=akey, secret_key=skey,
+                                            endpoint_url=S3_CFG["s3_url"])
+        return s3_mp_test_obj.simple_multipart_upload(bucket, obj, filesize, self.mp_obj_path, 4)
+
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_copy
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-46133')
+    def test_46133(self):
+        """
+        Test copy-object operation within same bucket and across different buckets for simple
+        object when max-object limit is set in user level quota
+        """
+        self.log.info("STARTED:Test copy-object operation within same bucket and across different"
+                      "buckets for simple object when max-object limit is set in user level quota")
+        test_cfg = self.csm_conf["test_41746"]
+        config_dict = {"enabled": True, "max_size": test_cfg["max_size"],
+                       "max_objects": test_cfg["max_objects"]}
```

Comment:
I don't see max_objects is set in test_41746 in test_rest_capacity.yaml.Or its taken from some other file?

Reply:
Updated